src/SFA.DAS.Commitments.Infrastructure.UnitTests/Services/ApprenticeshipInfoServiceMapperTests/WhenIMapAFramework.cs

```diff
@@ -0,0 +1,100 @@
+using KellermanSoftware.CompareNetObjects;
+using NUnit.Framework;
+using SFA.DAS.Apprenticeships.Api.Types;
+using SFA.DAS.Commitments.Infrastructure.Services;
+
+namespace SFA.DAS.Commitments.Infrastructure.UnitTests.Services.ApprenticeshipInfoServiceMapperTests
+{
+    [TestFixture]
+    public class WhenIMapFramework
+    {
+        private ApprenticeshipInfoServiceMapper _mapper;
+        private FrameworkSummary _framework;
+
+        [SetUp]
+        public void Arrange()
+        {
+            _mapper = new ApprenticeshipInfoServiceMapper();
+
+            _framework = new FrameworkSummary
+            {
+                Id = "1",
+                Title = "TestTitle",
+                FrameworkName = "TestFrameworkName",
+                PathwayName = "TestPathwayName",
+                Level = 1,
+                CurrentFundingCap = 1000, //this is to become redundant
+                EffectiveFrom = new DateTime(2017, 05, 01),
+                EffectiveTo = new DateTime(2020, 7, 31),
+                FundingPeriods = new List<FundingPeriod>
+                {
+                    new FundingPeriod { EffectiveFrom = new DateTime(2017,05,01), EffectiveTo = new DateTime(2018, 12, 31), FundingCap = 5000 },
+                    new FundingPeriod { EffectiveFrom = new DateTime(2019,01,01), EffectiveTo = new DateTime(2020, 7, 31), FundingCap = 2000 }
+                }
+            };
+        }
+
+        [Test]
+        public void ThenTitleIsMappedCorrectly()
+        {
+            //Act
+            var result = _mapper.MapFrom(new List<FrameworkSummary> { TestHelper.Clone(_framework) });
+
+            //Assert
+            var expectedTitle = $"{_framework.Title}, Level: {_framework.Level}";
+            Assert.AreEqual(expectedTitle, result.Frameworks[0].Title);
+        }
+
+        [Test]
+        public void ThenEffectiveFromIsMappedCorrectly()
+        {
+            //Act
+            var result = _mapper.MapFrom(new List<FrameworkSummary> { TestHelper.Clone(_framework) });
+
+            //Assert
+            Assert.AreEqual(_framework.EffectiveFrom, result.Frameworks[0].EffectiveFrom);
+        }
+
+        [Test]
+        public void ThenEffectiveToIsMappedCorrectly()
+        {
+            //Act
+            var result = _mapper.MapFrom(new List<FrameworkSummary> { TestHelper.Clone(_framework) });
+
+            //Assert
+            Assert.AreEqual(_framework.EffectiveFrom, result.Frameworks[0].EffectiveFrom);
+        }
+
+
+        [Test]
+        public void ThenFundingPeriodsAreMappedCorrectly()
+        {
+            //Act
+            var result = _mapper.MapFrom(new List<FrameworkSummary> { TestHelper.Clone(_framework) });
+
+            //Assert
+            var comparer = new CompareLogic(new ComparisonConfig
+            {
+                IgnoreObjectTypes = true
+            });
+
+            Assert.IsTrue(comparer.Compare(result.Frameworks[0].FundingPeriods, _framework.FundingPeriods).AreEqual);
```

Comment:
Your helper class, has a a method called EnumerablesAreEqual<T> which seems to do this. It seems to read better to use this method. I like your solution to clone objects for testing

Reply:
Updated to use the new EnumerablesAreEqual test helper method.